File: src/main/java/ru/ezhov/rocket/action/configuration/ui/RocketActionConfigurationRepository.java
package ru.ezhov.rocket.action.configuration.ui;

import ru.ezhov.rocket.action.api.RocketActionConfiguration;

import java.util.List;
import java.util.Optional;

public interface RocketActionConfigurationRepository {
    List<RocketActionConfiguration> all();

    Optional<RocketActionConfiguration> by(String type);
}

File: src/main/kotlin/ru/ezhov/rocket/action/configuration/domain/RocketActionConfigurationRepository.kt
package ru.ezhov.rocket.action.configuration.domain

import ru.ezhov.rocket.action.api.RocketActionConfiguration
import ru.ezhov.rocket.action.api.RocketActionType

interface RocketActionConfigurationRepository {
    fun all(): List<RocketActionConfiguration>
    fun by(type: RocketActionType): RocketActionConfiguration?
}